endpoint.go

```diff
@@ -417,36 +273,26 @@ func (ep *endpoint) Join(containerID string, options ...EndpointOption) error {
 		}
 	}()
 
-	err = ep.buildHostsFiles()
-	if err != nil {
+	address := ""
+	if ip := ep.getFirstInterfaceAddress(); ip != nil {
+		address = ip.String()
+	}
+	if err = sb.updateHostsFile(address, network.getSvcRecords()); err != nil {
 		return err
 	}
 
-	err = ep.updateParentHosts()
-	if err != nil {
+	if err = sb.updateDNS(ep.getNetwork().enableIPv6); err != nil {
 		return err
 	}
 
-	err = ep.setupDNS()
-	if err != nil {
+	if err = network.ctrlr.updateEndpointToStore(ep); err != nil {
 		return err
 	}
 
-	sb, err := ctrlr.sandboxAdd(sboxKey, !container.config.useDefaultSandBox, ep)
-	if err != nil {
-		return fmt.Errorf("failed sandbox add: %v", err)
-	}
-	defer func() {
-		if err != nil {
-			ctrlr.sandboxRm(sboxKey, ep)
-		}
-	}()
-
-	if err := network.ctrlr.updateEndpointToStore(ep); err != nil {
```

Comment:
Here it looks we were not failing the endpoint join if we fails to update the endpoint into the datastore (err not visible outside of if scope).
New changes took care of this.